DREAM3DReviewFilters/ReadBinaryCTNorthStar.cpp

```diff
@@ -206,69 +255,91 @@ int32_t ReadBinaryCTNorthStar::sanityCheckFileSizeVersusAllocatedSize(size_t all
       QString ss = QObject::tr("Binary file size is smaller than the number of allocated bytes");
-      setErrorCondition(-100, ss);
+      setErrorCondition(-38705, ss);
       return getErrorCode();
     }
 
-    FILE* f = fopen(m_InputFiles[iter].toLatin1().data(), "rb");
+    FILE* f = fopen(dataFileInput.first.toLatin1().data(), "rb");
     if(nullptr == f)
     {
-      QString ss = QObject::tr("Error opening binary input file: %1").arg(m_InputFiles[iter]);
-      setErrorCondition(-100, ss);
+      QString ss = QObject::tr("Error opening binary input file: %1").arg(dataFileInput.first);
+      setErrorCondition(-38706, ss);
       return getErrorCode();
     }
 
     ScopedFileMonitor monitor(f);
 
-    float* ptr = m_DensityPtr.lock()->getPointer(0);
-
+    size_t fileZSlice = 0;
     size_t index = 0;
-    size_t totalPoints = dims[0] * dims[1] * m_SlicesPerFile[iter];
-    int64_t progIncrement = static_cast<int64_t>(totalPoints / 100);
-    int64_t prog = 1;
-    int64_t progressInt = 0;
-    int64_t counter = 0;
 
-    for(size_t z = zShift; z < (zShift + m_SlicesPerFile[iter]); z++)
+    for(size_t z = zShift; z < (zShift + dataFileInput.second); z++)
     {
-      for(size_t y = 0; y < dims[1]; y++)
+      if(m_ImportSubvolume && (z < m_StartVoxelCoord[2] || z > m_EndVoxelCoord[2]))
       {
-        for(size_t x = 0; x < dims[0]; x++)
+        fileZSlice++;
+        continue;
+      }
+      QString ss = QObject::tr("Importing Data || Data File: %1 || Importing Slice %2").arg(dataFileInput.first).arg(z);
+      notifyStatusMessage(ss);
+      for(size_t y = 0; y < origDims[1]; y++)
+      {
+        if(m_ImportSubvolume && (y < m_StartVoxelCoord[1] || y > m_EndVoxelCoord[1]))
+        {
+          continue;
+        }
+
+        size_t fpOffset = ((origDims[1] * origDims[0] * fileZSlice) + (origDims[0] * y) + m_StartVoxelCoord[0]) * sizeof(float);
+        if(fseek(f, fpOffset, SEEK_SET) != 0)
         {
-          // Compute the proper index into the final storage array
-          index = (dims[0] * dims[1] * z) + (dims[0] * (dims[1] - y - 1)) + (dims[0] - x - 1);
-          std::ignore = fread(ptr + index, sizeof(float), 1, f);
+          QString ss = QObject::tr("Could not seek to postion %1 in file %2").arg(fpOffset).arg(dataFileInput.first);
+          setErrorCondition(-38707, ss);
+          fclose(f);
+          return getErrorCode();
+        }
 
-          // Check for progress
-          if(counter > prog)
-          {
-            progressInt = static_cast<int64_t>((static_cast<float>(counter) / totalPoints) * 100.0f);
-            QString ss = QObject::tr("Reading Data || Data File: %1 of %2 || %3% Completed").arg(iter + 1).arg(m_InputFiles.size()).arg(progressInt);
-            notifyStatusMessage(ss);
-            prog = prog + progIncrement;
-          }
-          counter++;
+        index = (importedDims[0] * importedDims[1] * (z - m_StartVoxelCoord[2])) + (importedDims[0] * (y - m_StartVoxelCoord[1])) + (0);
```

Comment:
Only major comment is here: the data are stored somewhat strangely in the dat files (I think YXZ or something to that effect), so this line is important to get the proper index.  Just wanted to check that this works for either use case of importing the whole volume or a subvolume.  Does this end up being the same equation as the original when importing a full volume?  

This was a really confusing aspect of the filter when we were writing it, so we essentially just checked based on whether the visualization looked correct.

Reply:
I checked visualizations of both full volumes and sub-volumes and both appeared correct.